src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -498,6 +504,8 @@ class ConstantPropagation extends Transform with ResolvedAnnotationPaths {
             case WRef(nodeName, _, NodeKind, _) if nodeMap.contains(nodeName) =>
               nodeRegCPEntries.getOrElseUpdate(nodeName, { regConstant(nodeMap(nodeName)) })
             case Mux(_, tval, fval, _) => regConstant(tval).resolve(regConstant(fval))
+            case DoPrim(Or, Seq(a, b), Nil, BoolType) => regConstant(Mux(a, one, b, BoolType))
+            case DoPrim(And, Seq(a, b), Nil, BoolType) => regConstant(Mux(a, b, zero, BoolType))
```

Comment:
Should we also be checking the alternative forms? eg. `Mux(b, one, a, ...)`?

Reply:
It checks these two since mux-const-prop puts them in that way. There's not really a great way to express "it's either one of these" without refactoring how reg const prop works entirely.

I'll add more cases that figure out which argument is a register. Since these are the only "special" arguments, this will make accomplish matching the alternative forms. I agree that this is a good idea since it decouples it from the `Mux` const prop implementation at the cost of two lines of code.

Reply:
I refactored the `regConstant` function to make it easier to try multiple orderings. This will also have the side effect of possibly optimizing away some **user** code that has similar `And`s and `Or`s.